### PR TITLE
ODO extension installer

### PR DESCRIPTION
**Pull request by GeekArthur**
Issue: https://github.com/eclipse/codewind/issues/451

This PR is for:
- Download ODO extension release on PFE build stage
- Install ODO extension and latest stable ODO CLI if the PFE is running on OpenShift/OKD cluster 

Signed-off-by: jingfu wang <jingfu.j.wang@ibm.com>

### CODE REVIEW

**Comment by GeekArthur** [comment]
@tobespc We need folks from Portal side to review this PR.
@elsony You can help review this PR as well.
Thanks all.

File: src/pfe/portal/modules/ExtensionList.js [hunk 1]
```diff
@@ -23,6 +23,8 @@ const log = new Logger(__filename);
 const extensionsDir = '/extensions';
 const extensionsPattern = /^(\S+)-(\d+\.\d+\.\d+)\.zip$/; // e.g. extension-name-0.0.1.zip
 const suffixOld = '__old';
+const odoExtensionName = "codewind-odo-extension";
+const odoBinarySource = "https://github.com/openshift/odo/releases/latest/download/odo-linux-amd64"
```
Comment (makandre) [inline src/pfe/portal/modules/ExtensionList.js]: Is the hardcoding of odo stuff in this file temporary?

Comment (GeekArthur) [reply src/pfe/portal/modules/ExtensionList.js]: We may or may not update/move this depends on the design/ODO CLI release. (eg. `https://github.com/openshift/odo/releases/latest/download/odo-linux-amd64` links to the latest stable ODO CLI, we don't need to change it for now, but may do some improvements for the installer)

Comment (makandre) [reply src/pfe/portal/modules/ExtensionList.js]: But I mean, we should not have "ODO" specific things hardcoded in Codwind.  I think @elsony there was an issue opened to further clean up hardcoding of extension specific things, so I was trying to confirm if this is also one of the items to be cleaned up later.

Do you guys have a link to that issue by any chance?

**Pull request merged**